### PR TITLE
Wip theming

### PR DESCRIPTION
**Pull request by David-Corbett**
progressive enhancements to the origins theme to add more components etc.

### CODE REVIEW

File: src/js/ext-win-script.js [hunk 1]
```diff
@@ -7,29 +7,39 @@
...
-  $("#main-content a[data-ext-type^='External']").each(function() {
-    $(this).attr('href', $(this).attr('data-ext-url'));
-    if ($(this).parent().hasClass('moreLink')) {
-      $(this).html('visit site');
+  Drupal.behaviors.nicsdruOriginsExtWin = {
+    attach: function (context) {
+      var $extLinkText = Drupal.t('external link opens in a new window / tab'),
+        $intLinkText = Drupal.t('opens in a new window / tab');
+
+      // External links - add identifiers
```
Comment (omahm) [inline src/js/ext-win-script.js]: full stop

File: src/js/ext-win-script.js [hunk 2]
```diff
@@ -7,29 +7,39 @@
...
+      $("#main-content a[href*='http://'],#main-content a[href*='https://'],#main-content a[href^='//']", context)
+        .once('elink').each(function () {
+        $(this).not('a.no-ext-icon, a:has(img), .social-links a')
+          .filter(function () {
+            return this.hostname && this.hostname !== location.hostname;
+          })
+          .append('<span class="visually-hidden">(' + $extLinkText + ')</span><svg aria-hidden="true" class="ico ico-elink"><title>' + $extLinkText + '</title><use xlink:href="#elink"></use></svg>')
+          .attr('title', $extLinkText)
+          .attr('target', '_blank')
+          .attr('rel', 'noopener noreferrer');
```
Comment (omahm) [inline src/js/ext-win-script.js:25]: 👍 good to see.

File: src/js/ext-win-script.js [hunk 3]
```diff
@@ -7,29 +7,39 @@
...
+      });
+
+      // Internal links with data-ext-url - turn them into external links
```
Comment (omahm) [inline src/js/ext-win-script.js]: full stop

File: src/js/ext-win-script.js [hunk 4]
```diff
@@ -7,29 +7,39 @@
...
+      $("#main-content a[data-ext-type^='External']", context).once('elink').each(function() {
+        $(this).attr('href', $(this).attr('data-ext-url'));
+        if ($(this).parent().hasClass('moreLink')) {
```
Comment (omahm) [inline src/js/ext-win-script.js]: Where is this class assigned? It doesn't follow the case convention.

File: src/scss/3_components/form/form.scss [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+select,
+[type='text'],
+[type='tel'],
+[type='url'],
+[type='email'],
+[type='password'],
+[type='date'],
+[type='time'],
+[type='number'],
+//input:not([type='button']):not([type='submit']):not([type='radio']):not([type='checkbox']):not([type='search']),
```
Comment (omahm) [inline src/scss/3_components/form/form.scss]: Remove commented out.

File: src/scss/3_components/form/form.scss [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  font-family: sans-serif;
+  font-weight: normal;
+  color: $charcoal;
+  line-height: 1.3;
+  padding: .6em 1.4em .5em .8em;
+  box-sizing: border-box;
+  margin: rhythm(1) 0 0;
+  border: 1px solid $grey-dark;
+  border-radius: 0;
+  -moz-appearance: none;
```
Comment (omahm) [inline src/scss/3_components/form/form.scss]: Prefixer should negate the need for these.

**Review by omahm — COMMENTED** [review]
Generally good. Need to stick to code standards for comments.

File: src/js/ext-win-script.js [hunk 7]
```diff
@@ -7,29 +7,39 @@
...
-  var $extLinkText = Drupal.t('external link opens in a new window / tab'),
-    $intLinkText = Drupal.t('opens in a new window / tab');
-  // Internal links with data-ext-url - turn them into external links
-  $("#main-content a[data-ext-type^='External']").each(function() {
-    $(this).attr('href', $(this).attr('data-ext-url'));
-    if ($(this).parent().hasClass('moreLink')) {
-      $(this).html('visit site');
+  Drupal.behaviors.nicsdruOriginsExtWin = {
+    attach: function (context) {
+      var $extLinkText = Drupal.t('external link opens in a new window / tab'),
```
Comment (johangant) [inline src/js/ext-win-script.js:12]: Just a small observation/question on `$` variable prefixes: looking at https://www.drupal.org/docs/develop/standards/javascript/javascript-coding-standards#variablesandarrays and above section, you'd usually only prefix with a `$` when you're expecting to handle a jQuery object.

File: src/js/ext-win-script.js [hunk 8]
```diff
@@ -7,29 +7,39 @@
...
+        $intLinkText = Drupal.t('opens in a new window / tab');
+
+      // External links - add identifiers
+      $("#main-content a[href*='http://'],#main-content a[href*='https://'],#main-content a[href^='//']", context)
```
Comment (johangant) [inline src/js/ext-win-script.js]: More nit picking :)

Separate parameters with a space after the comma for improved readability.

**Review by johangant — APPROVED** [review]
Looks decent, but guessing from title also ongoing? I've left some small comments to draw your attention to but nothing show-stopping in there.

Will mark as approved to allow you to merge as/when required.

File: src/scss/3_components/form/form.scss (hunk 5 above)
Comment (David-Corbett) [reply src/scss/3_components/form/form.scss]: haven't fully finished with this file, I'll remove - it was left to remind me what the original declaration was.

File: src/js/ext-win-script.js (hunk 3 above)
Comment (David-Corbett) [reply src/js/ext-win-script.js]: fixed

File: src/js/ext-win-script.js (hunk 1 above)
Comment (David-Corbett) [reply src/js/ext-win-script.js]: fixed

File: src/scss/3_components/form/form.scss (hunk 6 above)
Comment (David-Corbett) [reply src/scss/3_components/form/form.scss]: indeed. As per last, this file is a work in progress so a lot to still sort on it, this was just a basic start to help me with bugs around the search form on health condition pages

File: src/js/ext-win-script.js (hunk 4 above)
Comment (David-Corbett) [reply src/js/ext-win-script.js]: changed to the d8 version of the class

**Pull request merged**